### ISSUE TITLE
Python version conflict for dependencies

### ISSUE THREAD

**Issue by spectram** [PeterKamphuis/pyHIARD#13]
Casa modular is only available for python 3.8 [see this](https://casadocs.readthedocs.io/en/stable/notebooks/introduction.html#id1). But pyHIARD uses [importlib.resources.files()](https://docs.python.org/3/library/importlib.resources.html#importlib.resources.files) which was introduced in python 3.9.

I tried two fixes to [this code block](https://github.com/PeterKamphuis/pyHIARD/blob/dee679001a91becd9b8f4e2aedab1980c0daa859/pyHIARD/AGC/AGC.py#L124):
`data = pkg_resources.resource_string('pyHIARD', 'Templates/Input.fits')`

and

```
datafile = pkg_resources.resource_filename('pyHIARD', 'Templates/Input.fits')
shutil.copy(datafile,cfg.general.main_directory+'/Input.fits')
```
Both copy the 'Input.fits' to the general directory but the header is corrupted in a weird way. NAXIS1=500 and other bizarre issues. I have no explanations for this yet. Straight forward fix would be to read the fits file and write it to the new directory. Any thoughts on this?

**Comment by PeterKamphuis** [comment]
I thought these lines https://github.com/PeterKamphuis/pyHIARD/blob/dee679001a91becd9b8f4e2aedab1980c0daa859/pyHIARD/AGC/AGC.py#L35 should take care of this by importing importlib_resources instead of importlib.resources:

And input fits is a dummy file for which the NAXIS1 = 500 is correct as it is set in that same code block that you mentioned https://github.com/PeterKamphuis/pyHIARD/blob/dee679001a91becd9b8f4e2aedab1980c0daa859/pyHIARD/AGC/AGC.py#L124

Did you have an actual error or bad final cubes running in < python 3.8?  For me the code runs fine on python 3.6.

**Comment by spectram** [comment]
I am running it in Python 3.8.15, The importlib.resources module is available but the importlib.resources.files() function is new in python 3.9. This results in `AttributeError: module 'importlib.resources' has no attribute 'files'`. 

My above suggestions can fix this error but I noticed that the new Input.fits generated is 240 MB as opposed to ~ 15kB like the one in the Templates folder. This is the reason why I concluded that something bizarre must be happening.

Besides this, the package runs fine in the serial mode for me. Parallel mode results in an [Attribute error](https://stackoverflow.com/questions/45720153/python-multiprocessing-error-attributeerror-module-main-has-no-attribute).

But I am guessing that has more to do with my conda environment than the package itself. I am running it in a terminal and not an IPython console and so I have not found a solution yet.

**Comment by PeterKamphuis** [comment]
Yeah the Input.fits file is basically a lazy way of making a full radio header (with actual synthesis keywords)/template fits file. So for distribution it is made a very small cube which is then brought to an template of a typical size when copied.

This part does not use multiprocessing though, and the multiprocessing of the casa corruption is done by casa not by this code. I run everything in terminal as well but try to avoid conda and only use virtual environments. I'll try with one later with python 3.8.15.

**Comment by PeterKamphuis** [comment]
This should now all be solved in version 1.1.9.
For python versions > 3.8 the pip install will simply skip the the modular casa as otherwise it it will revert to version 1.1.2 as the last compatible version.
